src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -269,50 +298,99 @@ function entropy_projection!(cache, u, mesh::VertexMappedMesh, equations, dg::DG
+                  dim, u_local, equations, sparsity_pattern)
+  end
+end
+
+# Todo: DGMulti. Dispatch on curved/non-curved mesh types, this code only works for affine meshes (accessing rxJ[1,e],...)
+# When the operators are sparse, we use the sum-factorization approach to
+# computing flux differencing.
+@inline function local_flux_differencing!(fluxdiff_local, u_local, element_index,
+                                          flux_is_symmetric, volume_flux,
+                                          has_sparse_operators::Val{true}, mesh,
+                                          equations, dg, cache)
+  @unpack Qrst_skew, sparsity_patterns = cache
+  for dim in eachdim(mesh)
+    # There are two ways to write this flux differencing discretization on affine meshes.
+    # 1. Use numerical fluxes in Cartesian directions and sum up the discrete derivative
+    #    operators per coordinate direction accordingly.
+    # 2. Use discrete derivative operators per coordinate direction and corresponding
+    #    numerical fluxes in arbitrary (non-Cartesian) space directions.
+    #
+    # The first option can be implemented using
+    #
+    #   Q_skew = build_lazy_physical_derivative(e, dim, mesh, dg, cache)
+    #   hadamard_sum!(fluxdiff_local, Q_skew, volume_flux, dim,
+    #                 u_local, equations, sparsity_pattern)
+    #
+    # with `sparsity_pattern === cache.sparsity_pattern`.
+    # However, this option makes it necessary to sum up the individual
+    # `sparsity_patterns` of each reference coordinate direction. On tensor-product
+    # elements such as `Quad()` or `Hex()` elements, this increases the number of
+    # potentially expensive numerical flux evaluations by a factor of `ndims(mesh)`.
+    # Thus, we use the second option below (which basically corresponds to the
+    # well-known sum factorization on tensor product elements).
+    # Note that there is basically no difference for dense derivative operators.
+    normal_direction = get_contravariant_vector(element_index, dim, mesh)
+    sparsity_pattern = sparsity_patterns[dim]
+    Q_skew = Qrst_skew[dim]
+
+    # Use a `sparsity_pattern` to dispatch `hadamard_sum!`.
+    # For all elements, `sparsity_pattern::AbstractSparseMatrix{Bool}`.
+    hadamard_sum!(fluxdiff_local, Q_skew,
+                  flux_is_symmetric, volume_flux,
+                  normal_direction, u_local, equations, sparsity_pattern)
+  end
+end
+
+function calc_volume_integral!(du, u, volume_integral,
+                               mesh::VertexMappedMesh,
                                have_nonconservative_terms::Val{false}, equations,
                                dg::DGMultiFluxDiff{<:SBP}, cache)
 
   @unpack fluxdiff_local_threaded, sparsity_patterns, inv_wq, Qrst_skew = cache
   @unpack volume_flux = volume_integral
 
-  # Todo: DGMulti. Dispatch on curved/non-curved mesh types, this code only works for affine meshes (accessing rxJ[1,e],...)
+  flux_is_symmetric = Val{true}()
```

Comment:
Maybe rewrite this using `Val{true}()` and a comment when calling `local_flux_differencing!` as in the polynomial version below?